sympy/solvers/recurr.py

```diff
@@ -822,3 +822,126 @@ def rsolve(f, y, init=None):
             solution = solution.subs(result)
 
     return solution
+
+
+def rfind(seq):
```

Comment:
I'm ok with something like this function.  But you should think about naming.

Also, I think it's a bad idea to use hard-coded Symbol's for the output.  You can make optional arguments for this.